app/legacy_lib/export_recurring_donations.rb

```diff
@@ -2,7 +2,7 @@
 
 module ExportRecurringDonations
 
-  def self.initiate_export(npo_id, params, user_ids, requested_by_user = true)
+  def self.initiate_export(npo_id, params, user_ids, export_type = :requested_by_user)
```

Comment:
I'm not sure I understand what `:requested_by_user` should mean here. Now that I look at the original, I don't know what `requested_by_user` means in the original. It seems like means "do an export of all recurring donations". Is that correct do you think?

I think this should be changed in a way that is more descriptive of what the type of export actually is.

Reply:
Initially, the goal was to tell if it was an automatic report or if it was a report that the user requested through the UI. The reason for that is because the language changes on the email for the report: if it's a report requested by the user, then we send an email saying the report is available for download, otherwise, we give more information about which report is that (failed recurring donations from the past month? cancelled recurring donations from the past month?).

A boolean was working well for that when we had only two options, but now that we have three we need to describe the type of the report to decide which type of email we want to send.

What do you think about the types above?

* `requested_by_user_through_ui`;
* `failed_recurring_donations_automatic_report`;
* `cancelled_recurring_donations_automatic_report`.

Reply:
That would work!